src/components/tag/TagList.tsx

```diff
@@ -0,0 +1,40 @@
+import styled from "@emotion/styled";
+import { HTMLAttributes } from "react";
+import TagItem, { TagItemProps } from "components/tag/TagItem";
+import React, { useState } from "react";
+
+interface Props extends HTMLAttributes<HTMLDivElement> {
+  deletable?: boolean;
+}
+
+const TagList: React.FC<Props> = ({ children, deletable = false, ...props }: Props) => {
+  const [tagItems, setTagItems] = useState(children);
+
+  const handleTagItemClick = (index: number) => {
+    const newTagItems = React.Children.toArray(tagItems).filter((_, idx) => idx !== index);
+    setTagItems(newTagItems);
+  };
+
+  const tagItemsWithOnClick = React.Children.map(tagItems, (child, index) => {
+    if (deletable && React.isValidElement<TagItemProps>(child)) {
+      return React.cloneElement(child, {
+        onClick: () => handleTagItemClick(index),
+      });
+    }
+    return child;
+  });
+
+  return <EmotionWrapper {...props}>{tagItemsWithOnClick}</EmotionWrapper>;
+};
+
+export default TagList;
```

Comment:
Tag 관련해서 본 컴포넌트가 최상위 컴포넌트이고, TagItem만 단독으로 쓰일 여지가 없다면, 아래와 같이 변경하면 어떨까 합니다!
```suggestion
const Tags: React.FC<Props> = ({ children, deletable = false, ...props }: Props) => {
  const [tagItems, setTagItems] = useState(children);

  const handleTagItemClick = (index: number) => {
    const newTagItems = React.Children.toArray(tagItems).filter((_, idx) => idx !== index);
    setTagItems(newTagItems);
  };

  const tagItemsWithOnClick = React.Children.map(tagItems, (child, index) => {
    if (deletable && React.isValidElement<TagItemProps>(child)) {
      return React.cloneElement(child, {
        onClick: () => handleTagItemClick(index),
      });
    }
    return child;
  });

  return <EmotionWrapper {...props}>{tagItemsWithOnClick}</EmotionWrapper>;
};

export default Tags;
```

Reply:
넵 좋습니다! 파일 이름도 Tags로 바꾸겠습니다.